internal/cmd/rsl/record/record.go

```diff
@@ -12,20 +14,32 @@ type options struct{}
 
 func (o *options) AddFlags(_ *cobra.Command) {}
 
-func (o *options) Run(_ *cobra.Command, args []string) error {
+func (o *options) Run(cmd *cobra.Command, args []string) error {
 	repo, err := repository.LoadRepository()
 	if err != nil {
 		return err
 	}
 
-	return repo.RecordRSLEntryForReference(args[0], true)
+	if err := repo.RecordRSLEntryForReference(args[0], true); err != nil {
+		return err
+	}
+
+	if err := repo.VerifyRef(cmd.Context(), args[0], false); err != nil {
+		if err := repo.InvalidateLatestRSLEntry(); err != nil {
+			return err
+		}
+		return fmt.Errorf("verifcation of reference to be record failed, creation of RSL entry is not possible, err = %w", err)
+	}
+
+	return nil
 }
 
 func New() *cobra.Command {
 	o := &options{}
 	cmd := &cobra.Command{
 		Use:               "record",
 		Short:             "Record latest state of a Git reference in the RSL",
+		Long:              "Record latest state of a Git reference in the RSL. If the entry fails full ref verification on the recorded ref, the rsl entry is skipped",
```

Comment:
this needs to be updated. Also note that RSL is always all caps in docs.

Reply:
Fixed